js/main.js

```diff
@@ -40,5 +47,82 @@ function submitInfo(event) {
   $entryForm.reset();
 }
 
+// function to render entries
+function renderEntry(entry) {
+  // creates elements of DOM tree:
+  const $row = document.createElement('div');
+  $row.className = 'row entry-row';
+  const $column = document.createElement('div');
+  $column.className = 'column-full entry-column';
+  const $entryContainer = document.createElement('div');
+  $entryContainer.className = 'entry-container';
+  const $list = document.createElement('ul');
+  $list.className = 'journal-entry';
+  const $listItem = document.createElement('li');
+  $listItem.className = 'user-entry';
+  const $img = document.createElement('img');
+  $img.setAttribute('src', entry.url);
+  $img.className = 'entry-img';
+  const $textContainer = document.createElement('div');
+  $textContainer.className = 'text-container';
+  const $title = document.createElement('h2');
+  $title.className = 'entry-title';
+  $title.textContent = entry.title;
+  const $notes = document.createElement('p');
+  $notes.className = 'entry-notes';
+  $notes.textContent = entry.notes;
+
+  // append DOM nodes to build tree
+
+  $row.appendChild($column);
+  $column.appendChild($entryContainer);
+  $entryContainer.appendChild($list);
+  $list.appendChild($listItem);
+  $listItem.appendChild($textContainer);
+  $listItem.appendChild($img);
+  $textContainer.appendChild($title);
+  $textContainer.appendChild($notes);
+
+  // prepends DOM tree to list
+  $entriesList.appendChild($row);
+}
+
+// looping function for renderEntry function - iterates nextEntryId
+function arrayLoop(array) {
+  for (let i = 0; i < array.length; i++) {
+    const $entriesArray = array[i];
+    // eslint-disable-next-line no-unused-vars
+    const $renderedEntry = renderEntry($entriesArray);
+  }
+}
+
+// toggles "no entries" text as appropriate:
+function toggleNoEntries(event) {
+  const $entriesArray = data.entries;
+  if ($entriesArray.length > 0) {
+    $noEntryMessage.classList.add('hidden');
+  }
+}
+
 // submit event listener
+$entryForm.addEventListener('submit', toggleNoEntries);
 $entryForm.addEventListener('submit', submitInfo);
```

Comment:
You should only have 1 `submit` event listener on the form. `toggleNoEntries` gets called inside your `submit` callback function